9 delete comment by user

Jord000 opened the pull request:
Bin buttons added per user login to the comments
when bin is enacted it will reseed the comments. comment button added error functionality when empty
bin changes color when clicked so if slow loading you know when finished

johnhutcheon commented on __tests__/app.test.js:
```diff
@@ -234,23 +233,97 @@ describe('POST /api/articles/:article_id/comments', () => {
+describe('PATCH: /api/articles/:article_id', () => {
+  const incVotes = { inc_votes: 20 };
+  const wrongVote = { notcorrect: 'incorrect' };
+  const reduceVotes = { inc_votes: -200 };
```
Nice ordering at the top of this describe block 🙌

Keb84 reviewed (approved):
Really nice work. merge when ready 👍

johnhutcheon commented on __tests__/app.test.js:
```diff
@@ -234,23 +233,97 @@ describe('POST /api/articles/:article_id/comments', () => {
+              'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700',
+            votes: -200,
+          });
+        }
```
Could you just add one more test for if we send a request to a valid but non-existent path please?

johnhutcheon reviewed (changes requested):
Really nice work again Jordan. Just one test missing on this one. Take a look at my comment and add, then resubmit please. Thanks! 🙂

nickkagame reviewed (approved):
good work!

Jord000 committed:
Merge branch '7-vote-on-article' into 8-post-comment-to-article

Jord000 committed:
further development on the function of adding comment

Jord000 committed:
Merge branch '7-vote-on-article' into 8-post-comment-to-article

Jord000 committed:
add logout to nav bar with functionality to reset session storage

Jord000 committed:
submitting comments now reseeds the comment list and shows the newest comment submitted

Jord000 committed:
submit clears input box

Jord000 committed:
cleaned up author title

Jord000 committed:
move tools

Jord000 committed:
change comment seed to a globally available context
Bin buttons added per user login to the comments
when bin is enacted it will reseed the comments. comment button added error functionality when empty
bin changes color when clicked so if slow loading you know when finished

Jord000 committed:
Merge branch 'main' into 9-delete-comment-by-user

Jord000 committed:
added color to vote button for visual acknowledgement that the request is happening

nc-nick-jones reviewed (changes requested):
Good work jordan! There's an infinite loop in there.  Also there could be a little more feedback - text notification perhaps?

Jord000 committed:
duplicate button removed

nc-nick-jones reviewed (changes requested):
Nice work Jordan.  Though there's an infinite loop in there!  Also I think maybe you could give a user a bit more feedback ( like some text, colour coded in a way)

Jord000 committed:
fixed infinite loop

Jord000 committed:
added previous vote error function and used to build delete comment error handling ability

Jord000 committed:
error response improved with pop up text

Jord000 committed:
change delete comment request back now error handling confirmed
